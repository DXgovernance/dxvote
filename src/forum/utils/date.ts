import moment from 'moment';

export const formatTime = date => {
  const d = new Date(date);
  return moment(d).format('MMM dd');
};

