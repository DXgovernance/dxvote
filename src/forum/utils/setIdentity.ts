import Identities from 'orbit-db-identity-provider';
import { getResolver } from '@ceramicnetwork/3id-did-resolver';

// Connects the logged in Ceramic user to OrbitDB
export async function setIdentity({ orbitdb, registry }, { ceramic, threeId }) {
  if (orbitdb.identity.type !== 'did') {
    Identities.DIDIdentityProvider.setDIDResolver(getResolver(ceramic));
    const identity = await Identities.createIdentity({
      type: 'DID',
      didProvider: threeId.getDidProvider(),
    });
    orbitdb.identity = identity;
    registry.setIdentity(identity);
    console.log('OrbitDB: DID identity set!', identity);
  }
}
