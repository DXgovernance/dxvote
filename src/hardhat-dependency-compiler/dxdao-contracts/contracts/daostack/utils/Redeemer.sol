// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/daostack/utils/Redeemer.sol';
