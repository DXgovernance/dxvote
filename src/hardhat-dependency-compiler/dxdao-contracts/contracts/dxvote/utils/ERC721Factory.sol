// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/dxvote/utils/ERC721Factory.sol';
