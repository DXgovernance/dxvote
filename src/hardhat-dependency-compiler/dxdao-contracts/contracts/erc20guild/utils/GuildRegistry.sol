// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/erc20guild/utils/GuildRegistry.sol';
