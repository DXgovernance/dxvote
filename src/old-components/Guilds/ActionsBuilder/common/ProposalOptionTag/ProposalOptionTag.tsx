import React from 'react';
import styled from 'styled-components';
import { Option } from '../../types';

interface ProposalActionTagProps {
  option: Option;
}

const Tag = styled.span<ProposalActionTagProps>`
  margin: 0;
  border-radius: 0.375rem;
  padding: 0.25rem 0.5rem;
  border: 1px solid ${({ theme }) => theme.colors.muted};
  font-weight: ${({ theme }) => theme.fontWeights.medium};
  color: ${({ color }) => color};
  border-color: ${({ color }) => color};
`;

export const ProposalOptionTag: React.FC<ProposalActionTagProps> = ({
  option,
}) => {
  return <Tag color={option.color}>{option.label}</Tag>;
};
