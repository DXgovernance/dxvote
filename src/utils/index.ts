export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
