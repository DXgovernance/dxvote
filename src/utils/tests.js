import { ThemeProvider } from 'styled-components';
import { render as rtlRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { GuildsDarkTheme } from 'Components/theme';

export const renderWithRoutes = (ui, options = {}) => {
  //wrap with MemoryRouter
  return rtlRender(ui, { wrapper: MemoryRouter, ...options });
};

// we are mokcing testing-library/render with our own render
export const render = (children, options = {}) => {
  //wrap with Theme
  return renderWithRoutes(
    <ThemeProvider theme={GuildsDarkTheme}> {children} </ThemeProvider>,
    options
  );
};
