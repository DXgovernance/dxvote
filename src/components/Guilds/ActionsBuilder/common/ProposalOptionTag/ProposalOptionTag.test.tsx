import { render } from '../../../../../utils/tests';
import { ProposalOptionTag } from './ProposalOptionTag';

describe('ProposalOptionTag', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <ProposalOptionTag option={{ id: '1', color: 'blue', label: 'For' }} />
    );
    expect(container).toMatchSnapshot();
  });
});
