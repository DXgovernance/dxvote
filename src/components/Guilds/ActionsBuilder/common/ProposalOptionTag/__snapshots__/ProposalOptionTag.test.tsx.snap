// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalOptionTag Should match snapshot 1`] = `
<div>
   
  <span
    class="sc-bczRLJ frTcns"
    color="blue"
  >
    For
  </span>
   
</div>
`;
